From fc375cfe35c829e6acbae56e9ad3d47726a78d23 Mon Sep 17 00:00:00 2001
Message-ID: <fc375cfe35c829e6acbae56e9ad3d47726a78d23.1787144643.git.jdenemar@redhat.com>
From: Peter Krempa <pkrempa@redhat.com>
Date: Tue, 21 Jul 2026 14:54:50 +0200
Subject: [PATCH] qemuMonitorGuestPanicEventInfoFormatMsg: Directly return
 message
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Remove the 'ret' variable and format the panic string on a single line.

Signed-off-by: Peter Krempa <pkrempa@redhat.com>
Reviewed-by: Ján Tomko <jtomko@redhat.com>
(cherry picked from commit 3f41645e3d8bf7b589de58d2d5052123a4d6fe9c)

https://redhat.atlassian.net/browse/RHEL-242546
---
 src/qemu/qemu_monitor.c | 40 +++++++++++++++++-----------------------
 1 file changed, 17 insertions(+), 23 deletions(-)

diff --git a/src/qemu/qemu_monitor.c b/src/qemu/qemu_monitor.c
index 5ecee978f9..86702f6599 100644
--- a/src/qemu/qemu_monitor.c
+++ b/src/qemu/qemu_monitor.c
@@ -3922,43 +3922,37 @@ qemuMonitorSetBlockThreshold(qemuMonitor *mon,
 char *
 qemuMonitorGuestPanicEventInfoFormatMsg(qemuMonitorEventPanicInfo *info)
 {
-    char *ret = NULL;
-
     switch (info->type) {
     case QEMU_MONITOR_EVENT_PANIC_INFO_TYPE_HYPERV:
-        ret = g_strdup_printf("hyper-v: arg1='0x%llx', arg2='0x%llx', "
-                              "arg3='0x%llx', arg4='0x%llx', arg5='0x%llx'",
-                              info->data.hyperv.arg1, info->data.hyperv.arg2,
-                              info->data.hyperv.arg3, info->data.hyperv.arg4,
-                              info->data.hyperv.arg5);
+        return g_strdup_printf("hyper-v: arg1='0x%llx', arg2='0x%llx', arg3='0x%llx', arg4='0x%llx', arg5='0x%llx'",
+                               info->data.hyperv.arg1, info->data.hyperv.arg2,
+                               info->data.hyperv.arg3, info->data.hyperv.arg4,
+                               info->data.hyperv.arg5);
         break;
     case QEMU_MONITOR_EVENT_PANIC_INFO_TYPE_S390:
-        ret = g_strdup_printf("s390: core='%d' psw-mask='0x%016llx' "
-                              "psw-addr='0x%016llx' reason='%s'",
-                              info->data.s390.core,
-                              info->data.s390.psw_mask,
-                              info->data.s390.psw_addr,
-                              info->data.s390.reason);
+        return g_strdup_printf("s390: core='%d' psw-mask='0x%016llx' psw-addr='0x%016llx' reason='%s'",
+                               info->data.s390.core,
+                               info->data.s390.psw_mask,
+                               info->data.s390.psw_addr,
+                               info->data.s390.reason);
         break;
     case QEMU_MONITOR_EVENT_PANIC_INFO_TYPE_TDX:
         if (info->data.tdx.has_gpa)
-            ret = g_strdup_printf("tdx: error_code='0x%x' message='%s' "
-                                  "additional error information can be found "
-                                  "at gpa page: '0x%016llx'",
-                                  info->data.tdx.error_code,
-                                  info->data.tdx.message,
-                                  info->data.tdx.gpa);
+            return g_strdup_printf("tdx: error_code='0x%x' message='%s' additional error information can be found at gpa page: '0x%016llx'",
+                                   info->data.tdx.error_code,
+                                   info->data.tdx.message,
+                                   info->data.tdx.gpa);
         else
-            ret = g_strdup_printf("tdx: error_code='0x%x' message='%s'",
-                                  info->data.tdx.error_code,
-                                  info->data.tdx.message);
+            return g_strdup_printf("tdx: error_code='0x%x' message='%s'",
+                                   info->data.tdx.error_code,
+                                   info->data.tdx.message);
         break;
     case QEMU_MONITOR_EVENT_PANIC_INFO_TYPE_NONE:
     case QEMU_MONITOR_EVENT_PANIC_INFO_TYPE_LAST:
         break;
     }
 
-    return ret;
+    return NULL;
 }
 
 
-- 
2.55.0
